test/shopify_app/controller_concerns/login_protection_test.rb

```diff
@@ -172,6 +175,7 @@ class LoginProtectionControllerTest < ActionController::TestCase
   end
 
   test "#current_shopify_session is memoized and does not retreive session twice" do
+    ShopifyApp.configuration.allow_cookie_authentication = true
     with_application_test_routes do
       session[:shop_id] = "foobar"
```

Comment:
It doesn't seem like this needs to be cookie-based. It just needs to work one way or the other.

How about this?

```suggestion
    with_application_test_routes do
      request.env['jwt.shopify_domain'] = 'foobar'
```

You'd also need to update the `expects` to:
```ruby
ShopifyApp::SessionRepository.expects(:retrieve_shop_session_by_shopify_domain).returns(session).once
```

, but it lets us avoid changing config values.